frontend/catalyst/pennylane_extensions.py

```diff
@@ -167,13 +173,29 @@ def __call__(self, *args, **kwargs):
         )
 
         jaxpr = jax.make_jaxpr(self.fn)(*args)
-        if len(jaxpr.eqns) != 1:
-            raise TypeError("catalyst.grad can only be used on QNodes and other grad calls")
-        if jaxpr.eqns[0].primitive != jprim.func_p:
-            raise TypeError(
-                f"Attempting to differentiate something other than a function "
-                f"(got {jaxpr.eqns[0].primitive})"
-            )
+        assert len(jaxpr.eqns) == 1, "Expected jaxpr consisting of a single function call."
+        assert (
+            jaxpr.eqns[0].primitive == jprim.func_p
+        ), "Expected jaxpr consisting of a single function call."
```

Comment:
```suggestion
        assert len(jaxpr.eqns) == 1, "Expected jaxpr consisting of a single function call"
        assert (
            jaxpr.eqns[0].primitive == jprim.func_p
        ), "Expected jaxpr consisting of a single function call"
```

Reply:
After a quick look through it, I wouldn't say there is any consistency to the way assertions are currently formatted in the frontend 😅 Either way, removing the period while leaving it capitalized makes even less sense to me.
For an assertion this looks fine to me (the rule we have about punctuation/capitalization is for documentation strings).

Reply:
> there is any consistency to the way assertions are currently formatted in the frontend

That's bad! There should be some consistency but it may be not really related to this PR...

Let's get back to this if someone tries to do any frontend factoring again.